File: examples/exam1sharp/Album.cs
using System;
using Worm.Entities.Meta;
using Worm.Entities;
using Worm.Cache;
using Worm;

namespace test
{
    [Entity("dbo", "albums", "1")]
	class Album : KeyEntity
	{
		private string _name;
		private DateTime? _release;
        private int _id;

		public Album() {}

		public Album(int id, CacheBase cache, ObjectMappingEngine schema)
		{
            Init(id, cache, schema);
		}

        [EntityProperty("name")]
		public virtual string Name
		{
			get { using (Read("Name")) { return _name; } }
			set { using (Write("Name")) { _name = value; } }
		}

        [EntityProperty("release_dt")]
		public virtual DateTime? Release
		{
			get { using (Read("Release")) { return _release; } }
			set { using (Write("Release")) { _release = value; } }
		}

        [EntityProperty("id", Field2DbRelations.PrimaryKey, DBType="int")]
        public override object Identifier
        {
            get { return _id; }
            set { _id = (int)value; }
        }
    }
}

File: examples/exam1/Album.vb
Imports Worm.Entities
Imports Worm.Entities.Meta
Imports Worm
Imports Worm.Cache

Namespace test
    <Entity("dbo", "Albums", "1")> _
    Public Class Album
        Inherits KeyEntity

        Private _name As String
        Private _release As System.Nullable(Of Date)
        Private _id As Integer

        Public Sub New()

        End Sub

        Public Sub New(ByVal id As Integer, ByVal cache As CacheBase, ByVal schema As ObjectMappingEngine)
            Init(id, cache, schema)
        End Sub

        <EntityProperty("name")> _
        Public Overridable Property Name() As String
            Get
                Using Read("Name")
                    Return _name
                End Using
            End Get
            Set(ByVal value As String)
                Using Write("Name")
                    _name = value
                End Using
            End Set
        End Property

        <EntityProperty("release_dt")> _
        Public Overridable Property Release() As Nullable(Of Date)
            Get
                Using Read("Release")
                    Return _release
                End Using
            End Get
            Set(ByVal value As Nullable(Of Date))
                Using Write("Release")
                    _release = value
                End Using
            End Set
        End Property

        <EntityProperty("id", Field2DbRelations.PrimaryKey, DBType:="int")> _
        Public Overrides Property Identifier() As Object
            Get
                Return _id
            End Get
            Set(ByVal value As Object)
                _id = CInt(value)
            End Set
        End Property

    End Class
End Namespace